packages/frontend/src/services/StakingFarmContracts.js

```diff
@@ -46,4 +60,34 @@ export default class StakingFarmContracts {
             )
         );
     }
+
+    static isFarmingValidator(accountId) {
+        return (
+            getValidationVersion(
+                NODE_URL.indexOf(MAINNET) > -1 ? MAINNET : TESTNET,
+                accountId
+            ) === FARMING_VALIDATOR_VERSION
+        );
+    }
+
+    static hasUnclaimedRewards = async ({
+        contractName,
+        account_id,
+        from_index,
+        limit,
+    }) => {
+        return (
+            StakingFarmContracts.isFarmingValidator(contractName) &&
+            (await StakingFarmContracts.getFarmListWithUnclaimedRewards({
+                contractName,
+                account_id,
+                from_index,
+                limit,
+            }).then(
```

Comment:
```suggestion
            StakingFarmContracts.getFarmListWithUnclaimedRewards({
                contractName,
                account_id,
                from_index,
                limit,
            }.then(
```
`await` is unnecessary here